sdk/eventhubs/azure-messaging-eventhubs/test/eventhubs-stress-test/src/inc/argagg.hpp

```diff
@@ -963,12 +963,16 @@ inline bool flag_is_short(const char* s)
 
 inline bool parser_map::known_short_flag(const char flag) const
 {
-  return this->short_map[flag] != nullptr;
```

Comment:
Stupid question: Why does this fail to compile in eventhubs but the exact same code succeeds in core\perf?

THis file was copied from core\perf into eventhubs.

Reply:
Found it.

The correct fix for this is to add the following to cmakelists.txt:

```
if (MSVC)
  #argagg warnings
  target_compile_options(azure-perf PUBLIC /wd28020 /wd28204)
endif()
```
which is how core\perf works around the problem.

Reply:
I agree with your comment you made offline, it is better to suppress the warning via cmake pragmas, while keeping the code as close to the 3rd party code as possible.
FWIW, the current version of argagg (a change that was made 2 years ago, actually) does have static_casts in such places: https://github.com/vietjtnguyen/argagg/blob/78a0a9e9ae1dfc436809d2ac71479d8a4c2991a3/include/argagg/argagg.hpp#L1076 (not that it fixes Code Analysis; it still complains erroneously, until you unwind the code as I did)

Reply:
Fascinating. I may want to resync to the master at some point, if only to pick up bug fixes.